pkg/netpol/connlist/parsed_resources_test.go

```diff
@@ -0,0 +1,1304 @@
+	podObj.TypeMeta.APIVersion = "v1"
+	podObj.TypeMeta.Kind = "Pod"
+	podObj.ObjectMeta.Name = name
+	podObj.ObjectMeta.Namespace = namespace
+	podObj.Labels = map[string]string{"pod": name}
+	for _, port := range ports {
+		for _, protocol := range protocols {
+			podObj.Spec.Containers = append(podObj.Spec.Containers, newDefaultContainer(port, protocol))
+		}
+	}
+	return podObj
+}
+
+func newDefaultContainer(port int, protocol v1.Protocol) v1.Container {
+	contObj := v1.Container{}
+	contObj.Name = fmt.Sprintf("cont-%d-%s", port, strings.ToLower(string(protocol)))
+	contObj.Ports = make([]v1.ContainerPort, 1)
+	contObj.Ports[0].Name = fmt.Sprintf("serve-%d-%s", port, strings.ToLower(string(protocol)))
+	contObj.Ports[0].ContainerPort = int32(port)
+	contObj.Ports[0].Protocol = protocol
+	return contObj
+}
+
+type podInfo struct {
+	namespaces []string
+	podNames   []string
+	ports      []int
+	protocols  []v1.Protocol
+}
+
+type ParsedResourcesTest struct {
+	Name                   string
+	OutputFormat           string
+	ExpectedOutputFileName string
+	PodResources           podInfo
+	ANpList                []*v1alpha1.AdminNetworkPolicy
+	Banp                   *v1alpha1.BaselineAdminNetworkPolicy
+	NpList                 []*netv1.NetworkPolicy
+	TestInfo               string
+	Resources              []parser.K8sObject
+}
+
+func addMetaData(meta *metav1.ObjectMeta, cnt *int) {
+	if meta.Name == "" {
+		meta.Name = fmt.Sprintf("generated_name_%q", *cnt)
+		*cnt++
+	}
+	if meta.Namespace == "" {
+		meta.Namespace = "default"
+	}
+}
+
+func (test *ParsedResourcesTest) InitParsedResourcesTest() {
+	var genCnt = 0
+	test.TestInfo = fmt.Sprintf("test: %q, output format: %q", test.Name, test.OutputFormat)
+	for _, ns := range test.PodResources.namespaces {
+		parsedNs := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns, Labels: map[string]string{"ns": ns}}}
+		k8sObj := parser.CreateNamespaceK8sObject(parsedNs)
+		test.Resources = append(test.Resources, k8sObj)
+		for _, pod := range test.PodResources.podNames {
+			parsedPod := newDefaultPod(ns, pod, test.PodResources.ports, test.PodResources.protocols)
+			k8sObj := parser.CreatePodK8sObject(&parsedPod)
+			test.Resources = append(test.Resources, k8sObj)
+		}
+	}
+	for _, np := range test.NpList {
+		addMetaData(&np.ObjectMeta, &genCnt)
+		k8sObj := parser.CreateNetwordPolicyK8sObject(np)
+		test.Resources = append(test.Resources, k8sObj)
+	}
+	for _, anp := range test.ANpList {
+		addMetaData(&anp.ObjectMeta, &genCnt)
+		k8sObj := parser.CreateAdminNetwordPolicyK8sObject(anp)
+		test.Resources = append(test.Resources, k8sObj)
+	}
+	// Tanya: uncomment the code below when BaselineAdminNetworkPolicy is implemented
+	// if test.Banp != nil {
+	//  addMetaData(&test.Banp.ObjectMeta, &gen_cnt)
+	// 	k8sObj := parser.CreateBaselineAdminNetwordPolicyK8sObject(test.Banp)
+	// 	test.Resources = append(test.Resources, k8sObj)
+	// }
+}
+
+func stringForEval(peer eval.Peer) string {
```

Comment:
better use the current API for `Peer` instead of changing it with exposing `eval.PeerIPToIPBlock`.
can use both `peer.String()` and `peer.IP()`, and return the relevant (non empty) one.

Reply:
After separating `eval` tests from `connlist` expected results, the above function is no longer needed.